app/org/sagebionetworks/bridge/play/controllers/BaseController.java

```diff
@@ -258,13 +260,25 @@ void verifySupportedVersionOrThrowException(Study study) throws UnsupportedVersi
                 LOG.debug("Malformed Accept-Language header sent: " + acceptLanguageHeader);
             }
         }
+
+        // if no Accept-Language header detected, we shall add an extra warning header
+        Http.Response response = Http.Context.current().response();
+        response.setHeader(BridgeConstants.BRIDGE_API_STATUS_HEADER, BridgeConstants.BRIDGE_WARNING_STATUS);
+
         return new LinkedHashSet<>();
     }
     
     ClientInfo getClientInfoFromUserAgentHeader() {
         String userAgentHeader = request().getHeader(USER_AGENT);
         ClientInfo info = ClientInfo.fromUserAgentCache(userAgentHeader);
-        
+
+        // if the user agent cannot be parsed (probably due to missing user agent string or unrecognizable user agent),
+        // should set an extra header to http response as warning - we should have an user agent info for filtering to work
+        if (info.equals(ClientInfo.UNKNOWN_CLIENT)) {
+            Http.Response response = Http.Context.current().response();
+            response.setHeader(BridgeConstants.BRIDGE_API_STATUS_HEADER, BridgeConstants.BRIDGE_WARNING_STATUS);
```

Comment:
Yes, but need to include an explanation of what is wrong, something we can log from the client. Otherwise, we don't know what the exact problem is as an end user.